spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/trace/StackdriverTraceAutoConfiguration.java

```diff
@@ -118,8 +132,19 @@ public ManagedChannel stackdriverSenderChannel() {
 				.build();
 	}
 
-	@Bean
-	@ConditionalOnMissingBean
```

Comment:
Any pointers why we are not conditional here any more?

Reply:
No you're correct, that needs to be `@ConditionalOnMissingBean(name = REPORTER_BEAN_NAME)` and `@ConditionalOnMissingBean(name = SENDER_BEAN_NAME)`.

So the idea behind this is that whenever you add a tracing system (GCP Trace, Zipkin or smth else) to your classpath you'll always need to have a `Reporter<Span>` and `Sender`.
Behind the scenes the `TraceAutoConfiguration` will create a composite so you can send your traces to multiple systems "at the same time". (quoted because it's async)
That's also the reason why those 2 beans need to have a fixed name: to be able to override the autoconfig, otherwise you'll send your traces twice to the same system.